File: crud_react/src/services/EmployeeService.js
import axios from "axios";

const EMPLOYEE_API_BASE_URL = "http://localhost:8080/api/v1/employees";

class EmployeeService{

    createEmployee(employee){
        return axios.post(EMPLOYEE_API_BASE_URL, employee);
    }
    getEmployees(){
        return axios.get(EMPLOYEE_API_BASE_URL);
    }
    getEmployeeById(employeeId){
        return axios.get(EMPLOYEE_API_BASE_URL + '/' + employeeId);
    }
    updateEmployee(employee, employeeId){
        return axios.put(EMPLOYEE_API_BASE_URL + '/' + employeeId, employee);
    }

    searchEmployee(searchTerm){
        return axios.get(EMPLOYEE_API_BASE_URL + '/search/' + searchTerm);
    }

    deleteEmployee(employeeId){
        return axios.delete(EMPLOYEE_API_BASE_URL + '/' + employeeId);
    }

    

}
export default new EmployeeService();

File: crudSpring/src/main/java/com/ali/crudSpring/service/EmployeeService.java
package com.ali.crudSpring.service;

import com.ali.crudSpring.model.Employee;

import java.util.List;

public interface EmployeeService {
    Employee createEmployee(Employee employee);

    List<Employee> getAllEmployees();

    boolean deleteEmployee(Long id);

    Employee getEmployeeById(Long id);

    Employee updateEmployee(Long id, Employee employee);

//    public List<Employee> getEmployeesByFirstName(String firstName) ;

    public List<Employee> searchEmployees(String searchTerm) ;

    }